File: Benchmarks/src/main/java/benchmarks/rxjava/Simple.java
package benchmarks.rxjava;

import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.BenchmarkMode;
import org.openjdk.jmh.annotations.Fork;
import org.openjdk.jmh.annotations.Measurement;
import org.openjdk.jmh.annotations.Mode;
import org.openjdk.jmh.annotations.OutputTimeUnit;
import org.openjdk.jmh.annotations.Scope;
import org.openjdk.jmh.annotations.State;
import org.openjdk.jmh.annotations.Threads;
import org.openjdk.jmh.annotations.Warmup;
import rx.Observable;

import java.util.concurrent.TimeUnit;

@BenchmarkMode(Mode.Throughput)
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@Warmup(iterations = 5, time = 1000, timeUnit = TimeUnit.MILLISECONDS)
@Measurement(iterations = 5, time = 100, timeUnit = TimeUnit.MILLISECONDS)
@Fork(1)
@Threads(8)
@State(Scope.Benchmark)
public class Simple {
	Observable<Integer> source = Observable.just(5);

	@Benchmark
	public void mapping() {
		source.map(s -> s + 1).map(s -> s + 1).map(s -> s + 1);
	}
}

File: Benchmarks/src/main/scala/benchmarks/rxscala/Simple.scala
package benchmarks.rxscala

import org.openjdk.jmh.annotations._
import java.util.concurrent.TimeUnit
import rx.lang.scala.Observable

@BenchmarkMode(Array(Mode.Throughput))
@OutputTimeUnit(TimeUnit.MILLISECONDS)
@Warmup(iterations = 5, time = 1000, timeUnit = TimeUnit.MILLISECONDS)
@Measurement(iterations = 5, time = 100, timeUnit = TimeUnit.MILLISECONDS)
@Fork(1)
@Threads(8)
@State(Scope.Benchmark) class Simple {

  var source: Observable[Integer] = Observable.just(5)

  @Benchmark def mapping(): Unit = {
    source.map(_ + 1).map(_ + 1).map(_ + 1)
  }
}
